packages/logs/src/domain/logger.ts

```diff
@@ -38,6 +45,37 @@ export const HandlerType = {
 export type HandlerType = (typeof HandlerType)[keyof typeof HandlerType]
 export const STATUSES = Object.keys(StatusType) as StatusType[]
 
+export const StatusMapping: { [key in StatusType]: { match: (v: string) => boolean; priority: number } } = {
+  [StatusType.emerg]: { priority: 0, match: (v) => matchList(['f', 'emerg'], v, true) },
+  [StatusType.alert]: { priority: 1, match: (v) => matchList(['a'], v, true) },
+  [StatusType.critical]: { priority: 2, match: (v) => matchList(['c'], v, true) },
+  [StatusType.error]: { priority: 3, match: (v) => matchList(['e'], v, true) && !matchList(['emerg'], v, true) },
+  [StatusType.warn]: { priority: 4, match: (v) => matchList(['w'], v, true) },
+  [StatusType.notice]: { priority: 5, match: (v) => matchList(['n'], v, true) },
+  [StatusType.info]: { priority: 6, match: (v) => matchList(['i'], v, true) },
+  [StatusType.debug]: { priority: 7, match: (v) => matchList(['d', 'trace', 'verbose'], v, true) },
+  [StatusType.OK]: { priority: 8, match: (v) => matchList(['o', 's'], v, true) || matchList(['ok', 'success'], v) },
+}
+
+const remap = (rawStatus: string | number): StatusType => {
+  if (includesItem(STATUSES, rawStatus)) {
+    return rawStatus
+  }
+
+  for (const status of STATUSES) {
+    const { match, priority } = StatusMapping[status]
+    if (typeof rawStatus === 'string') {
+      if (match(rawStatus.toLowerCase())) {
+        return status
+      }
+    } else if (rawStatus >= 0 && rawStatus <= 7 && rawStatus === priority) {
+      return status
+    }
+  }
```

Comment:
**Suggestion:** ‏I think this approach is quite inefficient. Something link this could be more perfomant, and maybe a bit less verbose:

```ts
const statusByPriority = Object.values(StatusMapping) // we should reorder `StatusMapping` to be in sorted by priority
const statusByPrefix = {
  f: StatusType.emerg,
  emerg: StautsType.emerg,
  ...
}

if (typeof rawStatus === 'string') {
  rawStatus = rawStatus.toLowerCase()
  for (const prefixLength of [7, 5, 1]) { // optim: those are the only prefix lengths
    const prefix = rawStatus.slice(0, prefixLength)
    if (Object.prototype.hasOwnProperty.call(statusByPrefix, prefix)) {
      return statusByPrefix[prefix]
    }
  }
} else {
  return statusByPriority[rawStatus]
}
```

Reply:
Nice! I've push a version where I still use the `StatusMapping` because there's the case of "error" where we need to ensure that the `rawValue` is not a prefix if "emerg" (which starts with an 'e').